rust/cache/src/foyer.rs

```diff
@@ -131,6 +136,14 @@ pub struct FoyerCacheConfig {
     #[serde(default = "default_recover_concurrency")]
     pub recover_concurrency: usize,
 
+    /// Enable deterministic hashing.
+    /// If true, the cache will use a deterministic hasher which is stable
+    /// across restarts. Note that this hasher is not necessarily stable across
+    /// architectures or versions of foyer, and the underlying AHash.
+    #[arg(long, default_value_t = true)]
+    #[serde(default = "default_deterministic_hashing")]
+    pub deterministic_hashing: bool,
```

Comment:
(Not related to your changes) I feel that there could be two separate config types - one for in memory and another for cache. By using the same type aka `FoyerCacheConfig`, we can't enforce semantics such as `deterministic_hashing=false` only makes sense for in-memory cache and not disk. Similarly some fields above disk cache capacity, etc do not make sense for in memory cache.

Reply:
I agree